### PR TITLE
Redesigned Facility Home Page

### LINKED ISSUE #3713: Blue color of phone number makes the facility page UI less uniform
**Describe the bug**
Blue color of phone number makes the facility page UI less uniform

**To Reproduce**
 Go to http://localhost:4000/facility/eeb23cb7-1524-479a-825e-660311386b47

**Expected behavior**
Instead we can have a call icon and make the text black in color

**Screenshots**
image

### LINKED ISSUE #3757: Redesign Facility Home Page
Tasks / components:

- [x] Update style for `ButtonV2` when `style="secondary"` to match figma.
- [x] `FeatureButton`
- [x] `Menu` (dropdown) https://headlessui.com/react/menu
- [x] `PhoneNumber`
- [x] `Table`
- [x] `SummarySection` which should take in title (string / JSX Element) and options (JSXElement[]), which will be used to compose Oxygen Information section, Bed Capacity section, etc..
- [x] `BedCapacityCard`
- [x] `DoctorTypeCard`

### LINKED ISSUE #3804: Make the facility detail page tab left aligned
**Describe the bug**

currently, all the icons are centrally aligned

**To Reproduce**

https://care.coronasafe.in/facility/5092d953-6742-49b0-b1e3-025c53a89e8a

**Expected behavior**

make it left aligned to have uniformity

**Screenshots**

image

### LINKED ISSUE #4007: Patient dashboard: Text overflowing in buttons
**Describe the bug**
When the side nav menu is expanded, there is not enough space for text in buttons on top right of the page

Screenshot 2022-11-09 at 10 53 03 AM

Fix the spacing

### PR DESCRIPTION
**Pull request by GokulramGHV**
fixes #3757 
fixes #3804 
fixes #4007
fixes #3713 

## Proposed Changes

- Redesigned Facility Home Page

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Comment by nihal467** [comment]
@GokulramGHV since it is a redesign, marking it as team discussion required, for final approval of design in EOD call

**Comment by nihal467** [comment]
@GokulramGHV 
image

- [ ] triage is not responsive in mobile view

image
- [ ] in the oxygen table is not responsive

File: src/style/index.css
Comment (rithviknishad) [inline src/style/index.css]: @GokulramGHV revert this, and use the new ButtonV2 with type/style="secondary" instead of adjusting classes.

File: src/Components/Common/components/Menu.tsx
Comment (rithviknishad) [inline src/Components/Common/components/Menu.tsx]: Replace this with already implemented `DropdownTransition` since we need consistent dropdown animation for all types of dropdowns, which is already a wrapper around headlessui's transition.

File: src/Components/Facility/DoctorsCountCard.tsx
Comment (rithviknishad) [inline src/Components/Facility/DoctorsCountCard.tsx]: use unicons or move svg's to constants and make it tailwind compatible instead of hardcoding the colors.

File: src/Components/Facility/DoctorsCountCard.tsx
Comment (rithviknishad) [inline src/Components/Facility/DoctorsCountCard.tsx]: if these color's are not there in our existing tailwind color palette, add it to tailwind config similar to how patient colors were added. 
image

**Comment by GokulramGHV** [comment]
@nihal467 I've made the changes that you had mentioned

> @GokulramGHV image
> 
> * [ ]  triage is not responsive in mobile view
> 
> image
> 
> * [ ]  in the oxygen table is not responsive

File: src/Components/Common/components/Menu.tsx
Comment (rithviknishad) [inline src/Components/Common/components/Menu.tsx]: You can lock the design flexibility by using variants (similar to how button v2 is implemented) so that we achieve consistent design across the app but provide design flexibility based on the context.

1. Accept `variant: ButtonVariant` prop instead of `activeColor` and default it to `"primary".
image

2. Apply colours styles based on the variant
You might be able to directly use the `button-[primary/secondary/...]-default` classes along and append `hover:bg-opacity-10` to achieve the style.
image
image

**Review by rithviknishad — COMMENTED** [review]
@GokulramGHV I meant for the DropdownItem, not the menu button 😅. 
image

**Comment by GokulramGHV** [comment]
> @GokulramGHV I meant for the DropdownItem, not the menu button 😅. 
> image
> 

Oh sorry my bad 😅

**Comment by rithviknishad** [comment]
## New changes made

- Improve how ButtonV2 and DropdownMenuItem authorization works
- Unauthorized actions are hidden in dropdown menu:
image
image

**Comment by nihal467** [comment]
@GokulramGHV see a comment by rithvik, once completed, change it to need testing

**Comment by GokulramGHV** [comment]
> @GokulramGHV see a comment by rithvik, once completed, change it to need testing

@nihal467  I've made all the changes that rithvik suggested. The PR is ready for testing.

**Comment by nihal467** [comment]
@GokulramGHV after the comments, i can't see any pushes to the PR, are you sure, all the changes are been merged to the PR

**Comment by rithviknishad** [comment]
@nihal467  the last comment by me was just about the changes that I have pushed. Not a request for changes to be made. It's ready for testing

**Comment by nihal467** [comment]
@GokulramGHV the PR looks good to me. Marking it as team discussion required for the final approval of the PR

**Below suggestion only to be worked after todays discussion if got approved** 

image

the number highlighted in blue, doesn't look good, is there any other better alternative for it

**Comment by nihal467** [comment]
@GokulramGHV use the default blue colour for the phone number

**Comment by GokulramGHV** [comment]
@nihal467 @aparnacoronasafe @rithviknishad Is this color okay for the phone no.?
image

**Comment by rithviknishad** [comment]
This looks fine I feel. Can you also add `tracking-widest` or `tracking-wider` class to the phone number? Phone numbers would be easier to read.

Also lets wrap that to a component `ContactLink` which has props:
```tsx
type ContactLinkProps =
  | { mailto: string; tel: undefined }
  | { mailto: undefined; tel: string };
```

This way we can re-use the component with a consistent design everywhere in future.

File: src/Components/Common/components/ContactLink.tsx
Comment (rithviknishad) [inline src/Components/Common/components/ContactLink.tsx]:
```suggestion
            props.tel ? "uil-outgoing-call" : "uil-envelope-alt"
```

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
@GokulramGHV PR looks good to me. Marking it as "Team discussion required" for final approval

**Comment by nihal467** [comment]
PR final design approved, @khavinshankar it is good to merge

File: src/Components/Facility/BedTypeCard.tsx
Comment (khavinshankar) [inline src/Components/Facility/BedTypeCard.tsx]: can't we calculate the percentage directly without the useEffect, and why timeout?

File: src/Components/Facility/FacilityHome.tsx
Comment (khavinshankar) [inline src/Components/Facility/FacilityHome.tsx]: making a util function should make it bit cleaner, maybe a memoized function, your call!

File: src/Components/Facility/HospitalList.tsx
Comment (khavinshankar) [inline src/Components/Facility/HospitalList.tsx]: we are filtering it 2 times and repeating the code 3 times, can we export this into a function something like,

```js
      const getFacilityFeatureIcon = (featureId: number) => {
        const feature = FACILITY_FEATURE_TYPES.find(
          (f) => f.id === featureId
        );
  
        if (!icon) return null;
  
        return typeof feature.icon === "string" ? (
          <i
            className={feature.icon}
          />
        ) : (
          feature.icon
        )}{" "}
        &nbsp;
        {
          feature.name
        }
      }                       
```

**Review by khavinshankar — CHANGES_REQUESTED** [review]
@GokulramGHV some minor changes

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**
